platform-common-core/src/main/java/com/softicar/platform/common/core/number/formatter/BigDecimalScaleApplier.java

```diff
@@ -0,0 +1,92 @@
+package com.softicar.platform.common.core.number.formatter;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+/**
+ * Applies a given scale to {@link BigDecimal} values.
+ *
+ * @author Oliver Richers
+ */
+public class BigDecimalScaleApplier {
+
+	private Integer scale;
+	private RoundingMode roundingMode;
+
+	public BigDecimalScaleApplier() {
+
+		this.scale = null;
+		this.roundingMode = null;
+	}
+
+	/**
+	 * Defines the desired scale of the values, i.e. the number of decimal
+	 * places.
+	 * <p>
+	 * If <i>null</i> is provided as scale (which is the default), no changes
+	 * will be applied to the value.
+	 *
+	 * @param scale
+	 *            the scale; or <i>null</i> to disable scaling
+	 * @return this
+	 */
+	public BigDecimalScaleApplier setScale(Integer scale) {
+
+		this.scale = scale;
+		return this;
+	}
+
+	/**
+	 * Defines the {@link RoundingMode} for the values.
+	 * <p>
+	 * If <i>null</i> is provided as {@link RoundingMode} (which is the
+	 * default), no rounding will be done. Only redundant, trailing zeros will
+	 * be added or discarded from the fractional part to approximate the desired
+	 * scale as possible.
+	 *
+	 * @param roundingMode
+	 *            the {@link RoundingMode}; or <i>null</i> to disable rounding
+	 * @return this
+	 */
+	public BigDecimalScaleApplier setRoundingMode(RoundingMode roundingMode) {
+
+		this.roundingMode = roundingMode;
+		return this;
+	}
+
+	/**
+	 * Applies the scale defined by calling {@link #setScale(Integer)}.
```

Comment:
I think this Javadoc should at least mention "rounding" or "rounding mode".

Reply:
did so